components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -1437,7 +1438,50 @@ public static Time getTime(Double value, String unit) {
     }
     return new Time(value, UNITS.SECOND);
   }
-  
+
+
+  /**
+   * Formats the input value for the position into a length of the given unit.
+   *
+   * @param value  the value of the position
+   * @param unit   the unit of the position
+   *
+   * @return       the position formatted as a {@link Length}
+   */
+  public static Length getStagePosition(Double value, Unit<Length> unit) {
+    if (value == null || value.isInfinite()) {
+      LOGGER.debug("Expected float value for stage position; got {}", value);
+      return null;
+    }
+
+    if (unit == null) {
+      LOGGER.debug("Expected valid unit for stage position; got {}", unit);
+      return null;
+    }
+
+    return new Length(value, unit);
+  }
+
+  /**
+   * Formats the input value for the stage position into a length of the given
+   * unit.
+   *
+   * @param value  the value of the position
+   * @param unit   the unit of the position
+   *
+   * @return       the physical size formatted as a {@link Length}
+   */
+  public static Length getStagePosition(Double value, String unit) {
+      Unit<Length> baseunit = null;
+      try {
+        baseunit = UnitsLengthEnumHandler.getBaseUnit(
+          UnitsLength.fromString(unit));
+      } catch (EnumerationException e) {
+        LOGGER.debug(e.getMessage());
```

Comment:
The idea of the last commit was to mirror the behavior of `getStagePosition(Double, Unit<Length>)` where invalid units return `null`. Looking at it again, in the case of an `EnumerationException`, we might want to `return null` immediately after logging the exception.
Adding extra granularity to choose whether or not to fallback to the default unit is possible but I it would be good to clarify of the use cases before adding the API.